oxide-auth-actix/src/lib.rs

```diff
@@ -4,69 +4,457 @@
 //! `AsActor<_>` to create an actor implementing endpoint functionality via messages.
 #![warn(missing_docs)]
 
-mod future_endpoint;
-mod endpoint;
-pub mod message;
-pub mod request;
-#[cfg(test)]
-mod tests;
+use actix::{MailboxError, Message};
+use actix_web::{
+    dev::{HttpResponseBuilder, Payload},
+    error::BlockingError,
+    http::{
+        header::{
+            HeaderMap, InvalidHeaderValue, InvalidHeaderValueBytes, AUTHORIZATION, CONTENT_TYPE,
+            LOCATION, WWW_AUTHENTICATE,
+        },
+        HttpTryFrom, StatusCode,
+    },
+    web::Form,
+    web::Query,
+    FromRequest, HttpRequest, HttpResponse, Responder, ResponseError,
+};
+use futures::Future;
+use oxide_auth::{
+    endpoint::{Endpoint, NormalizedParameter, OAuthError, WebRequest, WebResponse},
+    frontends::{
+        dev::{Cow, QueryParameter},
+        simple::endpoint::Error,
+    },
+};
+use std::{error, fmt};
+use url::Url;
 
-use std::fmt;
-use std::error;
+mod operations;
 
-use actix_web::{HttpRequest, HttpResponse};
-use actix_web::ResponseError;
+pub use operations::{Authorize, Refresh, Resource, Token};
 
-// pub use self::endpoint::CodeGrantEndpoint;
-pub use oxide_auth::endpoint::{PreGrant, OAuthError, OwnerConsent, OwnerSolicitor};
-pub use oxide_auth::primitives::grant::Grant;
-pub use self::request::OAuthFuture;
-pub use self::request::OAuthRequest;
-pub use self::request::OAuthResponse;
+/// Describes an operation that can be performed in the presence of an `Endpoint`
+///
+/// This trait can be implemented by any type, but is very useful in Actor scenarios, where an
+/// Actor can provide an endpoint to an operation sent as a message.
+pub trait OxideOperation: Sized + 'static {
```

Comment:
Should be `OAuthOperation` for consistency with the other names.

Reply:
Or did you intend to have two different naming schemes?

Reply:
That wasn't intentional. I've migrated everything to OAuth[Thing]